spark-connector/src/main/java/com/datastrato/gravitino/spark/plugin/GravitinoDriverPlugin.java

```diff
@@ -0,0 +1,98 @@
+  private static final Logger LOG = LoggerFactory.getLogger(GravitinoDriverPlugin.class);
+
+  private GravitinoCatalogManager catalogManager;
+
+  @Override
+  public Map<String, String> init(SparkContext sc, PluginContext pluginContext) {
+    SparkConf conf = sc.conf();
+    String gravitinoUri = conf.get(GravitinoSparkConfig.GRAVITINO_URI);
+    String metalake = conf.get(GravitinoSparkConfig.GRAVITINO_METALAKE);
+    Preconditions.checkArgument(
+        StringUtils.isNoneBlank(gravitinoUri),
+        String.format("{}:{}, should not empty", GravitinoSparkConfig.GRAVITINO_URI, gravitinoUri));
+    Preconditions.checkArgument(
+        StringUtils.isNoneBlank(metalake),
+        String.format(
+            "{}:{}, should not empty", GravitinoSparkConfig.GRAVITINO_METALAKE, metalake));
+
+    catalogManager = GravitinoCatalogManager.createGravitinoCatalogManager(gravitinoUri, metalake);
+    catalogManager.loadCatalogsFromGravitino();
+    registerGravitinoCatalogs(conf, catalogManager.getGravitinoCatalogs());
+    registerSqlExtensions();
+    return Collections.emptyMap();
+  }
+
+  @Override
+  public void shutdown() {
+    if (catalogManager != null) {
+      catalogManager.close();
+    }
+  }
+
+  private void registerGravitinoCatalogs(SparkConf sparkConf, Map<String, Catalog> catalogs) {
+    catalogs.forEach(
+        (catalogName, catalog) -> {
+          String sparkCatalogConfigName = "spark.sql.catalog." + catalogName;
+          Preconditions.checkArgument(
+              sparkConf.contains(sparkCatalogConfigName) == false,
+              catalogName + " is already registered to SparkCatalogManager");
+          if (catalog.provider().equalsIgnoreCase("hive")) {
+            registerGravitinoHiveCatalog(sparkConf, catalog, sparkCatalogConfigName);
+            LOG.info("Register " + catalogName + " to Spark catalog manager");
+          } else {
+            LOG.info(
+                "Skip register "
+                    + catalogName
+                    + ", because "
+                    + catalog.provider()
+                    + "is not supported");
+          }
+        });
+  }
+
+  private void registerGravitinoHiveCatalog(
+      SparkConf sparkConf, Catalog catalog, String sparkCatalogConfigName) {
+    sparkConf.set(sparkCatalogConfigName, GravitinoHiveCatalog.class.getName());
+    Preconditions.checkArgument(
+        catalog.properties() != null, "Hive Catalog properties should not be null");
+    String metastoreUri =
+        catalog.properties().get(GravitinoSparkConfig.GRAVITINO_HIVE_METASTORE_URI);
+    Preconditions.checkArgument(
+        StringUtils.isNoneBlank(metastoreUri),
+        "Couldn't get "
+            + GravitinoSparkConfig.GRAVITINO_HIVE_METASTORE_URI
+            + " from catalog properties");
+    sparkConf.set(
+        sparkCatalogConfigName + "." + GravitinoSparkConfig.SPARK_HIVE_METASTORE_URI, metastoreUri);
+  }
+
+  // Todo inject Iceberg extensions
```

Comment:
Why do we need to inject Iceberg externsion?

Reply:
Iceberg extensions add Iceberg specify parser and optimizer rules, it's useful when support Iceberg catalog